.changes/unreleased/BUG FIXES-20241209-151045.yaml

```diff
@@ -0,0 +1,6 @@
+kind: BUG FIXES
+body: Fix incorrect validation for for-expressions
```

Comment:
Would love something more verbose here to indicate to the user the improvement

Reply:
"Fix incorrect validation errors in local blocks when using a for-expression"? I'm open to suggestions

Reply:
```suggestion
body: Fix incorrect validation of for-expressions by correctly detecting tuple and object expressions inside for-expressions
```

Reply:
> Fix incorrect validation of references by correctly detecting tuple and object expressions created from for-expressions

I would like to avoid "inside", because supporting completion inside the `[for ... ]` and `{for ... }` is a complex open issue